langchain/llms/beam.py

```diff
@@ -0,0 +1,299 @@
+                    name=self.name,
+                    cpu=self.cpu,
+                    memory=self.memory,
+                    gpu=self.gpu,
+                    python_version=self.python_version,
+                    python_packages=self.python_packages,
+                )
+            )
+
+    def run_creation(self) -> None:
+        """Creates a Python file which will be deployed on beam."""
+        script = textwrap.dedent(
+            """
+        import os
+        import transformers
+        from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+        model_name = "{model_name}"
+
+        def beam_langchain(**inputs):
+            prompt = inputs["prompt"]
+            length = inputs["max_length"]
+
+            tokenizer = GPT2Tokenizer.from_pretrained(model_name)
+            model = GPT2LMHeadModel.from_pretrained(model_name)
+            encodedPrompt = tokenizer.encode(prompt, return_tensors='pt')
+            outputs = model.generate(encodedPrompt, max_length=int(length),
+              do_sample=True, pad_token_id=tokenizer.eos_token_id)
+            output = tokenizer.decode(outputs[0], skip_special_tokens=True)
+
+            print(output)
+            return {{"text": output}}
+
+        """
+        )
+
+        script_name = "run.py"
+        with open(script_name, "w") as file:
+            file.write(script.format(model_name=self.model_name))
+
+    def _deploy(self) -> str:
+        """Call to Beam."""
+        try:
+            import beam  # type: ignore
+
+            if (beam.__path__) == "":
+                raise ImportError
+        except ImportError:
+            raise ValueError(
+                "Could not import beam python package. "
+                "Please install it with `curl "
+                "https://raw.githubusercontent.com/slai-labs"
+                "/get-beam/main/get-beam.sh -sSfL | sh`."
+            )
+
+        self.app_creation()
+        self.run_creation()
+
+        process = subprocess.run(
+            "beam deploy app.py", shell=True, capture_output=True, text=True
+        )
+
+        if process.returncode == 0:
+            print(process.stdout)
+            output = process.stdout
+            lines = output.split("\n")
+            app_id = None
+
+            for line in lines:
+                if line.startswith(" i  Send requests to: https://apps.beam.cloud/"):
+                    app_id = line.split("/")[-1]
+                    self.url = line.split(":")[1].strip()
+                    return app_id
+
+            raise ValueError(
+                f"""Failed to retrieve the appID from the deployment output.
+                Error: {process.stdout}"""
+            )
+        else:
+            raise ValueError(f"Deployment failed. Error: {process.stderr}")
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[list] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
```

Comment:
we should adhere to signature on abstract class, so no kwargs. could we pass in app_id at instantiation instead

Reply:
When I tried doing this it was throwing a linting error because it was overriding the LLM superclass. Happy to pass it at instantiation, if preferred.